services/harmony/test/jobs/jobs-status.ts

```diff
@@ -279,6 +310,52 @@ describe('Individual job status route', function () {
           expect(job.message).to.include('The job has completed successfully');
         });
 
+        it('supplies a link to the STAC catalog', function () {
+          const job = JSON.parse(this.res.text);
+          expect(hasSTACLink(job)).to.be.true;
+        });
+
+        itIncludesADataExpirationField();
+      });
+    });
+
+    describe('when the job has completed with errors', function () {
+      StubService.hook({
+        params: {
+          item: {
+            href: 'https://example.com',
+            type: 'image/gif',
+            bbox: '-10,-10,10,10',
+            temporal: '2020-01-01T00:00:00.000Z,2020-01-02T00:00:00.000Z',
+          },
+          status: 'complete_with_errors',
+          httpBackend: 'true',
+
+        },
+      });
+      hookRangesetRequest(version, collection, variableName, { username: 'jdoe4' });
+      before(async function () {
+        await this.service.complete();
+      });
+
+      describe('retrieving its job status', function () {
+        hookRedirect('jdoe4');
+
+        it('returns a status field of "complete_with_errors"', function () {
+          const job = JSON.parse(this.res.text);
+          expect(job.status).to.eql('complete_with_errors');
+        });
+
+        it('returns a human-readable message field corresponding to its state', function () {
+          const job = JSON.parse(this.res.text);
+          expect(job.message).to.include('The job has completed with errors. See the errors field for more details');
+        });
+
+        it('supplies a link to the STAC catalog', function () {
```

Comment:
I assume we have an existing test that verifies we do not have a STAC catalog link when the request is failed rather than complete_with_errors?

Reply:
Yeah, we do. But looking at it it probably passes for the wrong reason (checking the wrong field). Let me see if it needs to be fixed.